File: Components/RepositoryCategory/RepositoryCategoryController.vb
Imports System
Imports System.Configuration
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports System.Reflection
Imports System.Xml
Imports DotNetNuke
Imports DotNetNuke.Common.Utilities
Imports DotNetNuke.Common.Globals
Imports DotNetNuke.Services.Search

Namespace DotNetNuke.Modules.Repository
#Region "RepositoryCategory"

    Public Class RepositoryCategoryController

        Public Function GetRepositoryCategories(ByVal ModuleId As Integer, ByVal RootID As Integer) As ArrayList
            Return CBO.FillCollection(DataProvider.Instance().GetRepositoryCategories(ModuleId, RootID), GetType(RepositoryCategoryInfo))
        End Function
        Public Function GetSingleRepositoryCategory(ByVal ItemId As Integer) As RepositoryCategoryInfo
            Return CType(CBO.FillObject(DataProvider.Instance().GetSingleRepositoryCategory(ItemId), GetType(RepositoryCategoryInfo)), RepositoryCategoryInfo)
        End Function
        Public Function AddRepositoryCategory(ByVal ItemId As Integer, ByVal ModuleId As Integer, ByVal CategoryName As String, ByVal Parent As Integer, ByVal ViewOrder As Integer) As Integer
            Return CType(DataProvider.Instance().AddRepositoryCategory(ItemId, ModuleId, CategoryName, Parent, ViewOrder), Integer)
        End Function
        Public Sub UpdateRepositoryCategory(ByVal ItemId As Integer, ByVal CategoryName As String, ByVal Parent As Integer, ByVal ViewOrder As Integer)
            DataProvider.Instance().UpdateRepositoryCategory(ItemId, CategoryName, Parent, ViewOrder)
        End Sub
        Public Sub DeleteRepositoryCategory(ByVal ItemID As Integer)
            DataProvider.Instance().DeleteRepositoryCategory(ItemID)
        End Sub

    End Class

#End Region
End Namespace

File: Components/RepositoryCategory/RepositoryCategoryController.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Configuration;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using System.Reflection;
using System.Xml;
using DotNetNuke;
using DotNetNuke.Common.Utilities;
using DotNetNuke.Common;
using DotNetNuke.Services.Search;

namespace DotNetNuke.Modules.Repository
{
	#region "RepositoryCategory"

	public class RepositoryCategoryController
	{

		public ArrayList GetRepositoryCategories(int ModuleId, int RootID)
		{			
            var enumerable = CBO.FillCollection<RepositoryCategoryInfo>(DataProvider.Instance().GetRepositoryCategories(ModuleId, RootID));
            ArrayList arrayList = new ArrayList(enumerable);
            return arrayList;
		}
		public RepositoryCategoryInfo GetSingleRepositoryCategory(int ItemId)
		{
            return CBO.FillObject<RepositoryCategoryInfo>(DataProvider.Instance().GetSingleRepositoryCategory(ItemId));
		}
		public int AddRepositoryCategory(int ItemId, int ModuleId, string CategoryName, int Parent, int ViewOrder)
		{
			return Convert.ToInt32(DataProvider.Instance().AddRepositoryCategory(ItemId, ModuleId, CategoryName, Parent, ViewOrder));
		}
		public void UpdateRepositoryCategory(int ItemId, string CategoryName, int Parent, int ViewOrder)
		{
			DataProvider.Instance().UpdateRepositoryCategory(ItemId, CategoryName, Parent, ViewOrder);
		}
		public void DeleteRepositoryCategory(int ItemID)
		{
			DataProvider.Instance().DeleteRepositoryCategory(ItemID);
		}

	}

	#endregion
}
